crates/core/src/ring.rs

```diff
@@ -469,15 +469,20 @@ impl Ring {
     async fn acquire_new(
         &self,
         ideal_location: Location,
-        skip_list: &[&PeerId],
+        skip_list: &HashSet<&PeerId>,
         notifier: &EventLoopNotificationsSender,
         live_tx_tracker: &LiveTransactionTracker,
     ) -> anyhow::Result<Option<Transaction>> {
+        // Generate a new skip list that includes the already connected peers
+        let mut complete_skip_list: HashSet<PeerId> =
```

Comment:
I think this is a bug, if you add the conneted peers, who else are you supposed to query?

Instead, what you want is to pick a query target, and then update the skip list with all the connect peers, I think